pkg/kosmosctl/rsmigrate/options.go

```diff
@@ -0,0 +1,71 @@
+package rsmigrate
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	ctlutil "k8s.io/kubectl/pkg/cmd/util"
+
+	"github.com/kosmos.io/kosmos/pkg/apis/kosmos/v1alpha1"
+	kosmosversioned "github.com/kosmos.io/kosmos/pkg/generated/clientset/versioned"
+)
+
+type KnodeOptions struct {
+	KnodeName         string
+	Knode             *v1alpha1.Knode
+	KnodeNativeClient kubernetes.Interface
+	KnodeCrdClient    *kosmosversioned.Clientset
+}
+
+type CommandOptions struct {
+	MasterKubeConfig string
+	MasterClient     kubernetes.Interface
+	MasterCrdClient  *kosmosversioned.Clientset //clientset operate knode releted resource
+
+	SrcKnodeOptions *KnodeOptions
+	Namespace       string
+}
+
+func (o *CommandOptions) Validate(cmd *cobra.Command) error {
+
+	return nil
+}
+
+func (o *CommandOptions) Complete(f ctlutil.Factory, cmd *cobra.Command) error {
+	var masterConfig *restclient.Config
+	var err error
+	// get master kubernetes clientset
+	if len(o.MasterKubeConfig) > 0 {
+		masterConfig, err = clientcmd.BuildConfigFromFlags("", o.MasterKubeConfig)
+		if err != nil {
+			return fmt.Errorf("generate masterConfig failed: %s", err)
+		}
+	} else {
+		masterConfig, err = f.ToRESTConfig()
+		if err != nil {
+			return fmt.Errorf("generate masterConfig failed: %s", err)
+		}
+	}
+	crdClient, err := kosmosversioned.NewForConfig(masterConfig)
+	if err != nil {
+		return fmt.Errorf("create master knode clientset error: %s ", err)
+	}
+	o.MasterCrdClient = crdClient
+	masterClient, err := kubernetes.NewForConfig(masterConfig)
+	if err != nil {
+		return fmt.Errorf("create master clientset error: %s ", err)
+	}
+	o.MasterClient = masterClient
```

Comment:
Use the code in util to cleanup the code

Reply:
modified